sky/resources.py

```diff
@@ -1141,6 +1142,16 @@ def less_demanding_than(
                     return False
                 if self.accelerators[acc] > other_accelerators[acc]:
                     return False
+                if isinstance(self.accelerators[acc], float) or isinstance(
+                        other_accelerators[acc], float):
+                    # If the requested accelerator count is a float, we only
+                    # allow strictly equal counts since all of the float point
+                    # accelerator counts are less than 1 (e.g., 0.1, 0.5), and
+                    # we want to avoid semantic ambiguity (e.g. launching
+                    # with --gpus A10:0.25 on a A10:0.75 cluster).
+                    if not math.isclose(self.accelerators[acc],
+                                        other_accelerators[acc]):
+                        return False
```

Comment:
We should allow requested resources to be a float, while the existing accelerators to be a int, as long as requested resources is <= existing resources.

That said, the 
```suggestion
                if isinstance(other_accelerators[acc], float) and not other_accelerators[acc].is_integer():
                    # If the requested accelerator count is a float, we only
                    # allow strictly equal counts since all of the float point
                    # accelerator counts are less than 1 (e.g., 0.1, 0.5), and
                    # we want to avoid semantic ambiguity (e.g. launching
                    # with --gpus A10:0.25 on a A10:0.75 cluster).
                    if not math.isclose(self.accelerators[acc],
                                        other_accelerators[acc]):
                        return False
```

Reply:
Good point! Updated. Thanks!

Reply:
Actually, after a second thought, I think we should still keep the original `isinstance(self.accelerators[acc], float) or isinstance(other_accelerators[acc], float)` condition. Considering the following case: user submit the jobs with --gpus A10:0.5 and the cluster has A10:1. In fact the requirements 0.5 will be translated to 1 and thus the user can only have one A10:0.5 job instead of 2, which is confusing. The original condition capture such case but the updated one (`isinstance(other_accelerators[acc], float) and not other_accelerators[acc].is_integer()`) does not.

Reply:
We did allow having two `A10:0.5` running on a single cluster with `A10:1`. Do you know when did we change the behavior of this? or did we ever change this behavior before this PR?

Reply:
We need to fix this before we merge the PR